src/webgpu/shader/execution/memory_model/barrier.spec.ts

```diff
@@ -42,34 +43,92 @@ const memoryModelTestParams: MemoryModelTestParams = {
   numBehaviors: 2,
 };
 
+// The two kinds of non-atomic accesses tested.
+//  rw: read -> barrier -> write
+//  wr: write -> barrier -> read
+//  ww: write -> barrier -> write
+type AccessPair = 'rw' | 'wr' | 'ww';
+
+// Test the non-atomic memory types.
+const kMemTypes = [MemoryType.NonAtomicStorageClass, MemoryType.NonAtomicWorkgroupClass] as const;
+
 const storageMemoryBarrierStoreLoadTestCode = `
-  test_locations.value[x_0] = 1u;
+  test_locations.value[x_0] = 1;
   workgroupBarrier();
-  let r0 = test_locations.value[x_1];
+  let r0 = u32(test_locations.value[x_1]);
   atomicStore(&results.value[shuffled_workgroup * workgroupXSize + id_1].r0, r0);
 `;
 
 const workgroupMemoryBarrierStoreLoadTestCode = `
-  wg_test_locations[x_0] = 1u;
+  wg_test_locations[x_0] = 1;
   workgroupBarrier();
-  let r0 = wg_test_locations[x_1];
+  let r0 = u32(wg_test_locations[x_1]);
   atomicStore(&results.value[shuffled_workgroup * workgroupXSize + id_1].r0, r0);
 `;
 
+const storageMemoryBarrierLoadStoreTestCode = `
```

Comment:
nit: Could scope these inside the single function that uses them, to reduce global pollution.

Reply:
yeah, it becomes apparent after this restructuring.
There's not much else going on in this module, so I'll keep it.